packages/surimi/src/types/css-strict.types.ts

```diff
@@ -0,0 +1,164 @@
+import type csstype from 'csstype';
+
+import type { CustomProperty } from '#lib/api/custom-property';
+
+import type { VendorPrefix } from './config.types';
+
+/**
+ * Global CSS keywords that are valid for ANY CSS property.
+ * These are part of the CSS-wide keywords specification.
+ *
+ * @see https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Values_and_Units#css-wide_values
+ */
+export type GlobalCssValue = 'inherit' | 'initial' | 'unset' | 'revert' | 'revert-layer';
+
+/**
+ * Strict CSS properties with enhanced type hints.
+ * This type extends csstype properties with global CSS keywords for better autocomplete.
+ *
+ * Each property accepts:
+ * - Valid CSS values from csstype (includes all string values, numbers, etc.)
+ * - Global CSS keywords (inherit, initial, unset, revert, revert-layer)
+ * - CustomProperty instances with type hints
+ *
+ * Note: This type provides improved autocomplete and documentation but does not
+ * restrict invalid values at compile time due to TypeScript's structural typing.
+ * It's primarily intended to provide better developer experience through IDE hints.
+ *
+ * @example
+ * ```typescript
+ * const styles: StrictCssProperties = {
+ *   display: 'flex',     // ✅ Gets autocomplete
```

Comment:
The `autocomplete` part here is not really the selling point, right? Because you already get auto-complete for everything in the current version
<img width="661" height="281" alt="image" src="https://github.com/user-attachments/assets/d9f8c533-dfb7-4877-96e2-9e9339290fa0" />

It's just that currently, you can also pass all other strings as well. Which, as I stated in the summary, I think is important